### PR TITLE
Add deployment APIs at tenant scope

### PR DESCRIPTION
**Pull request by Tiano2017**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by daschult** [comment]
@openapibot sdkautomation rebuild

**Comment by daschult** [comment]
@openapibot sdkautomation rebuild

**Comment by daschult** [comment]
@openapibot sdkautomation rebuild

**Comment by NullMDR** [comment]
@openapibot sdkautomation rebuild

**Comment by yungezz** [comment]
@Tiano2017 there're errors in CI. pls fix them.
- modelvalidation error: https://dev.azure.com/azure-sdk/public/_build/results?buildId=74633&view=logs&j=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742. Error is MISSING_REQUIRED_PARAMETER

**Comment by Tiano2017** [comment]
@yungezz Looking at my example file: https://github.com/Azure/azure-rest-api-specs/pull/6720/files#diff-30a00039c98a25059cce3712940c7032, I see both parameters are provided. Not sure if there's anything I missed. Can you help me understand the error?

**Comment by yungezz** [comment]
@Tiano2017 try put `parameters` parameter into body as defined in swagger file. sample: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/authorization/resource-manager/Microsoft.Authorization/stable/2015-07-01/examples/PutRoleDefinition.json#L6

**Review by majastrz — APPROVED** [review]
Looks good to me.

**Comment by Tiano2017** [comment]
@yungezz I did pretty much the same as PutRoleDefinition, but it's still failing.. any more suggestions?

**Comment by Tiano2017** [comment]
@yungezz I tried a few approaches, but am still getting errors on model validation. Do you have any suggestions on what I need to do to fix them?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-07-01/resources.json [hunk 1]
```diff
@@ -0,0 +1,4502 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "No Content"
+          },
+          "404": {
```
Comment (yungezz) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2019-07-01/resources.json]: no need to add error code  in swagger file, pls add "default" back.

**Review by yungezz — CHANGES_REQUESTED** [review]
need to remove error code and add default

**Comment by majastrz** [comment]
>         },

We never respond with both validation results and error, right? (partial failure)

---
Refers to: specification/resources/resource-manager/Microsoft.Resources/stable/2019-07-01/resources.json:3788 in a18e248. [](commit_id = a18e2487d6fe269ea9a4a60dc250072da4276bc1, deletion_comment = True)

**Comment by Tiano2017** [comment]
@majastrz For export template we can return a response that contains both results and error. For other requests I don't think we do that.

**Comment by ravbhatnagar** [comment]
signing off on the changes added after the last sign off!

File: specification/resources/resource-manager/readme.md [hunk 2]
```diff
@@ -216,6 +216,15 @@ input-file:
 - Microsoft.Resources/stable/2019-05-10/resources.json
 ```
 
+### Tag: package-resources-2019-05
```
Comment (yungezz) [inline specification/resources/resource-manager/readme.md]: is this a typo?

**Review by yungezz — APPROVED** [review]

**Pull request merged**